### DISCUSSION TITLE
Diy proportional valve actuator

### DISCUSSION THREAD

**Discussion by nliaudat** [nliaudat/esp32_8ch_motor_shield#18] [Show and tell]
Get a look at https://github.com/nliaudat/floor-heating-proportional-valve

**Comment by nliaudat** [comment]
Need help for the new beta : https://github.com/nliaudat/floor-heating-proportional-valve/tree/beta

**Reply by nstadigs** [reply]
Impressive!

What's your motivation for giving every motor their own mcu? Are these made for distributed radiator valve control?

**Reply by nliaudat** [reply]
Yes.
The ESP32 C3 costs approx 1.9$ but the goal is to make a smart TRV valve, usable for standalone radiators and floor heating.
To be standalone, it will need a battery power source. For floor heating array, it'll only need an usb as power source

**Comment by rici44** [comment]
Hi. 

I like your project ESP32 8 channels motor controller. Proportional valve like that from Loxone https://shop.loxone.com/enen/valve-actuator-tree.html would help me to install underfloor heating for 12 zones. 

To know the exact possition of the valve is great. For radiator heating I boght Shelly TRV https://kb.shelly.cloud/knowledge-base/shelly-trv which shows percentage of openned valve 1-100%). 

As those mentioned are higher priced valves your idea is great and it would help to solve installations of underfloor heatings controlled by Home Assitent. 

I found one project with stepper motor https://www.instructables.com/Smart-Radiator-Valve-With-Home-Assistant/ and also one dismatled trv https://www.ifixit.com/Teardown/Tado+Smart+Radiator+Thermostat+v3+Plus+Teardown/129731. Second mentioned uses 25BYJ412-56 motor. I thoght stepper will rise the price but found some ready made stepper actuators https://www.alibaba.com/product-detail/New-design-micro-TRV-Linear-actuator_60642767668.html.

What do you think about stepper usage in trv? What are pros and cons?

How may I help you with your project?

**Reply by rici44** [reply]
Update: dismatled stepper with rodial to linear actuator 

https://www.micro-steppermotor.com/sale-11930934-25mm-3-2v-geared-stepper-motors-high-torque-with-rohs-certification-linear-stepper-motor25byj412l.html

**Comment by nliaudat** [comment]
I've studied the stepper motor solution, but steppers lost their position after a reboot. They also need 4 wires per motor instead of 2.

The best solution I found to get the position everytime is opto-IR sensor like in : https://github.com/nliaudat/floor-heating-proportional-valve.

In fact, the precise position is not very usefull in daily usage as the current algorithm process like that : 
Check temp > move actuator proportionnal to the diff temperature (the movements are small (<5%) as the temperature change is slow )
If movement hit endstop, no problem, the valve is at min/max and we just have to wait for the temp to adjust.

The Loxone valve seams to use a bus protocol like modbus or one-wire to chains. It's not only voltage.
Shelly TRV have wifi and do not need a controller.

Have a look at the https://github.com/Lenti84/VdMot_Controller , perhaps it fits better your need.
Regards

**Reply by rici44** [reply]
I need esphome which I understand a bit so arduino project at the https://github.com/Lenti84/VdMot_Controller is not the right one for my needs. 

Loxone is one-wire bus, so much more info inside their controller server.

Would not be helpfull the restore_value: True with stepper motor? Like this:

https://www.reddit.com/r/Esphome/comments/predqd/keeping_track_of_motor_steps_between_reboots/

https://github.com/RoadkillUK/Motor-on-a-Roller-Blind-for-ESPHOME

**Reply by nliaudat** [reply]
I've tried "restore_value" some times ago which was interesting instead of doing a calibration at each boot. 
I had no success with it. I'll get another try
